### PR TITLE
Add sbp callback for time message

### PR DESCRIPTION
**Pull request by anth-cole**
Adding sbp gps time message callback to RTCM3 to SBP decoder

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## v1.1.0-branch-26-geafe511
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-26-geafe511/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.1.0-branch-26-geafe511/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-26-geafe511

**Comment by anonymous-travis-swiftnav** [comment]
## v1.1.0-branch-27-g516ce44
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-27-g516ce44/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.1.0-branch-27-g516ce44/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-27-g516ce44

File: package/sbp_rtcm3_bridge/src/sbp_rtcm3_bridge.c [hunk 1]
```diff
@@ -88,6 +89,11 @@ int main(int argc, char *argv[])
     exit(EXIT_FAILURE);
   }
 
+  if(sbp_callback_register(SBP_MSG_GPS_TIME,gps_time_callback,NULL) != 0){
```
Comment (gsmcmullin) [inline package/sbp_rtcm3_bridge/src/sbp_rtcm3_bridge.c]: Add spaces after `if` and after commas, and before the opening brace

File: package/sbp_rtcm3_bridge/src/sbp_rtcm3.c [hunk 2]
```diff
@@ -491,4 +497,14 @@ void wgsecef2llh(const double ecef[3], double llh[3]) {
   llh[0] = copysign(1.0, ecef[2]) * atan(S / (e_c * C));
   llh[2] = (p * e_c * C + fabs(ecef[2]) * S - WGS84_A * e_c * A_n) /
            sqrt(e_c * e_c * C * C + S * S);
-}
\ No newline at end of file
+}
+
+void gps_time_callback(u16 sender_id, u8 len, u8 msg[], void *context) {
```
Comment (gsmcmullin) [inline package/sbp_rtcm3_bridge/src/sbp_rtcm3.c]: open brace on new line

File: package/sbp_rtcm3_bridge/src/sbp_rtcm3.c [hunk 3]
```diff
@@ -17,13 +17,20 @@
...
 #include <libsbp/observation.h>
+#include <libsbp/navigation.h>
 #include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
 
 // time given from rover observation data. Must be maintained to within half a week of
 // the epoch of incoming RTCM data
-static gps_time_nano_t time_from_rover_obs = { 1942, 0 };
+static gps_time_nano_t time_from_rover_obs = { .tow = 0, .ns = 0, .wn = .0 };
```
Comment (gsmcmullin) [inline package/sbp_rtcm3_bridge/src/sbp_rtcm3.c]: Explicit initialiser isn't needed.  The C runtime will set to zero.

File: package/sbp_rtcm3_bridge/src/sbp_rtcm3.c [hunk 4]
```diff
@@ -175,17 +182,16 @@ u8 rtcm3_obs_to_sbp(const rtcm_obs_message *rtcm_obs, msg_obs_t *sbp_obs[4],
           }
 
           sbp_obs[sbp_msg]->header.t.tow = rtcm_obs->header.tow * 1000.0;
-
-          if( sbp_obs[sbp_msg]->header.t.tow - time_from_rover_obs.tow < -302400 ) {
+          double timediff = gps_diff_time(&sbp_obs[sbp_msg]->header.t, &time_from_rover_obs);
+          if( timediff < -302400000 ) {
```
Comment (gsmcmullin) [inline package/sbp_rtcm3_bridge/src/sbp_rtcm3.c]: add a space after `if` and remove the spaces inside the brackets, same below.

File: package/sbp_rtcm3_bridge/src/sbp_rtcm3.c [hunk 5]
```diff
@@ -175,17 +182,16 @@ u8 rtcm3_obs_to_sbp(const rtcm_obs_message *rtcm_obs, msg_obs_t *sbp_obs[4],
...
               sbp_obs[sbp_msg]->header.t.wn = time_from_rover_obs.wn + 1;
           }
-          else if( sbp_obs[sbp_msg]->header.t.tow - time_from_rover_obs.tow > 302400 ) {
+          else if( timediff > 302400000 ) {
```
Comment (gsmcmullin) [inline package/sbp_rtcm3_bridge/src/sbp_rtcm3.c]: put else on the same line is the closing brace

File: package/sbp_rtcm3_bridge/src/sbp_rtcm3_bridge.c (hunk 1 above)
Comment (anth-cole) [inline package/sbp_rtcm3_bridge/src/sbp_rtcm3_bridge.c]: done

File: package/sbp_rtcm3_bridge/src/sbp_rtcm3.c (hunk 5 above)
Comment (anth-cole) [inline package/sbp_rtcm3_bridge/src/sbp_rtcm3.c]: done

File: package/sbp_rtcm3_bridge/src/sbp_rtcm3.c (hunk 2 above)
Comment (anth-cole) [inline package/sbp_rtcm3_bridge/src/sbp_rtcm3.c]: done

File: package/sbp_rtcm3_bridge/src/sbp_rtcm3.c (hunk 4 above)
Comment (anth-cole) [inline package/sbp_rtcm3_bridge/src/sbp_rtcm3.c]: done

**Comment by anonymous-travis-swiftnav** [comment]
## v1.1.0-branch-28-g07441ba
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-28-g07441ba/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.1.0-branch-28-g07441ba/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-28-g07441ba

**Comment by anonymous-travis-swiftnav** [comment]
## v1.1.0-branch-27-g2fe4745
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-27-g2fe4745/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.1.0-branch-27-g2fe4745/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-27-g2fe4745

**Comment by anonymous-travis-swiftnav** [comment]
## v1.1.0-branch-28-g1881b78
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-28-g1881b78/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.1.0-branch-28-g1881b78/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-28-g1881b78

**Pull request merged**